downloads/swan-lake-release-notes/swan-lake-alpha/RELEASE_NOTE.md

```diff
@@ -0,0 +1,1562 @@
+
+###### Socket
+
+The `socket` module was removed and got replaced by the `TCP` and `UDP` modules.
+
+#### Observability
+
+- A new extension model, which separates each extension into a separate module is introduced.
+    - Observability is not included into the final JAR by default. It can be added with the following configuration in the `Ballerina.toml`
+
+      ```toml
+      [build-options]
+      observabilityIncluded=true
+      ```
+
+    - The observability extension can be packaged by adding an import to the module in the code as shown in the example below.
+
+      ```ballerina
+      import ballerinax/prometheus as _;
+      ```
+
+- `Prometheus` and `Jaeger` extensions are introduced back.
+    - The `Prometheus` extension can be enabled by adding the following configuration in the `Config.toml` file.
+
+      ```toml
+      [ballerina.observe]
+      metricsEnabled=true
+      metricsReporter="prometheus"
+      
+      [ballerinax.prometheus]
+      host="127.0.0.1"  # Optional Configuration. Default value is localhost
+      port=9797         # Optional Configuration. Default value is 9797
+      ```
+
+    - The `Jaeger` extension can be enabled by adding the following config.
+
+      ```toml
+      [ballerina.observe]
+      tracingEnabled=true
+      tracingProvider="jaeger"
+      
+      [ballerinax.jaeger]
+      agentHostname="127.0.0.1"  # Optional Configuration. Default value is localhost
+      agentPort=6831             # Optional Configuration. Default value is 6831
+      ```
+
+    - By default, the `Jaeger` extension now publishes traces to the Jaeger Agent using the `jaeger.thrift` over the compact Thrift protocol.  
+
+#### Code to Cloud
+
+- The Kubernetes artifacts can be generated using the `--cloud=k8s` build option. The `import ballerina/cloud as _` is no longer required.
+- VS code plugin support for the `Kubernetes.toml`
+    - Code Completion based on the c2c specification
+    - Code Actions add/modify the probes and environments based on the source code.
+
+#### Breaking Changes
+
+- Resource method declarations are no longer allowed in object-type descriptors.
+- Resource methods are not considered to be part of the type.
+- Non-`isolated` service variables defined outside an `isolated` function can be accessed within the function only if the variable is a `final` variable and the type is a subtype of `readonly`.
+- The `@icon` annotation has been replaced with the `@display` annotation.
+- The value type of the XML iteration, which was previously `xml|string` is now `xml`. Moreover, the value type of the `xml<T>` iteration is now `T`. 
+
+#### Taint Analyzer Update
+
+With this release, the taint analyzer does not produce taint errors unless explicitly enabled. However, the taint analyzer does perform the taint flow analysis without producing errors regardless of it being enabled or not.
```

Comment:
Is the second sentence useful to the user? The user can't do anything about it, right? @rdhananjaya

Reply:
I felt it's incomplete to just say enable/disable taint analyzer. It's just a bit of information.

Reply:
If we are not going to produce errors, is there any point in running the taint analyzer?

Reply:
@rdhananjaya

Reply:
Yes, say module B depends on module A. 
Module A authors may not enable taint checking but if module B authors or who ever compiles that module enable taint checking we need the taint tables for functions from module A in order to derive taint tables for module B.

Reply:
@MaryamZi @pubudu91

Reply:
So technically what we enable/disable is logging the errors, right?

Shouldn't this be
```suggestion
With this release, the taint analyzer does not produce taint errors unless explicitly enabled. However, the taint analyzer still performs the taint flow analysis without producing errors if error logging is not enabled.
```